### DISCUSSION TITLE
Feature request: add keyboard shortcuts for moving active tab left / right

### DISCUSSION THREAD

**Discussion by cebaa** [doublecmd/doublecmd#1154] [Suggestions]
Tabs can currently be rearranged with mouse by dragging a tab. It would be useful to add keyboard shortcuts to do this as well. I use this daily with ConEmu, which by default has [these shortcuts assigned](https://conemu.github.io/en/KeyboardShortcuts.html):

Win+Alt+Left, Win+Alt+Right | Move active tab leftward/rightward (aka reorder ConEmu tabs)

**Comment by Skif-off** [comment]
@alexx2000 
we have `cm_MoveTabLeft` / `cm_MoveTabRight`, so should be closed as "complete"?

**Reply by dazu89** [reply]
So after some, back and forth, I figured that the function that  I was looking for, i.e. to move a tab from the left to the right pane and vice-versa, is not implemented. The clostest is called `cm_Exchange` (CTRL+U) which swaps to open tabs.
This feature request comes up every now and then, for example here on the forum: https://doublecmd.h1n.ru/viewtopic.php?t=5367
There is currently a work around for windows: https://github.com/Galicarnax/dcvim
That said, I think this is a must have for a double pane file manager!

**Reply by Skif-off** [reply]
@dazu89, try [quasimovetab.lua](https://github.com/j2969719/doublecmd-plugins/blob/master/scripts/lua/quasimovetab.lua).

**Reply by dazu89** [reply]
Thank you, @Skif-off, this is how it is solved in dcvim. Could you remind me, how to import the toolbar button definition from the .lua file? I have not used doublecmd in a while.

**Reply by Skif-off** [reply]
Just copy copy to the clipboard > right click in an empty area of the toolbar > paste ([help files](https://doublecmd.github.io/doc/en/toolbar.html#toolbarconfig)).
Don't forget to change the path to the script (the first parameter in the button).

**Comment by cebaa** [comment]
@Skif-off Thanks - and agreed, cm_MoveTabLeft/Right is what I was after. Closing this now.
